client/client.go

```diff
@@ -949,15 +950,31 @@ func (c *Client) fingerprint() error {
 			return err
 		}
 
-		c.configLock.Lock()
-		applies, err := f.Fingerprint(c.config, c.config.Node)
-		c.configLock.Unlock()
+		request := &cstructs.FingerprintRequest{Config: c.config, Node: c.config.Node}
+		response := &cstructs.FingerprintResponse{
+			Attributes: make(map[string]string, 0),
+			Links:      make(map[string]string, 0),
+			Resources:  &structs.Resources{},
+		}
+
+		err = f.Fingerprint(request, response)
 		if err != nil {
 			return err
 		}
-		if applies {
+
+		// if an attribute should be skipped, remove it from the list which we will
+		// later apply to the node
+		for _, e := range skipped {
```

Comment:
These aren't attributes that are being skipped. These are whole fingerprinters so they won't have created any attribute that needs to be deleted.

Reply:
Ah, only visible after expanding to get context that skipped is constructed from fingerprinter names. The diff made it look like it was mutating the output based on the skipped list.